src/xdocs/config_imports.xml

```diff
@@ -1112,6 +1123,20 @@ public class InputIllegalImport { }
   &lt;/subpackage&gt;
 &lt;/import-control&gt;
         </source>
+        <p>
+          In the example below, only classes that end with "Panel", "View", or "Dialog" in the
+          package <code>gui</code> are allowed to have imports from <code>javax.swing</code>. All
+          other classes in the package are disallowed.
+        </p>
+        <source>
+&lt;import-control pkg=&quot;com.puppycrawl.tools.checkstyle.checks&quot;&gt;
+  &lt;subpackage name=&quot;gui&quot;&gt;
+    &lt;class name=&quot;.*(Panel|View|Dialog)&quot; regex=&quot;true&quot;&gt;
+      &lt;allow pkg=&quot;javax.swing&quot;/&gt;
+    &lt;/class&gt;
```

Comment:
We need to show how to disallow in *Presentation and *Module classes

Reply:
What classes/packages do you want to disallow in those classes? awt and swing?

Reply:
In my mind, most of use cases for import control module is definition of black list (list of disallow).
In most applications there are 80-90% of code is closed for re-usage outside, in this part of code all is allowed to make it work. In some projects there are api part or any other part that is reused by others. In such small area users will likely to enforce rules on "white list of allowances" to control what could go outside (extra dependencies for others).

I am skipping out config from consideration as it is too fanatic.

So I try to place to example doc some use-case that is known for others, so user will find it and understand what is it for.
Probably it is better to make it like:
com.mycompany.billing
  *Panel,*View,*Dialog - disallow "com.mycompany.dao",. ".*.jdbc." usage.
  *PresentationModel - disallow javax,swing,J*  usage